app/controllers/admin_controller.rb

```diff
@@ -0,0 +1,25 @@
+class AdminController < ApplicationController
+  before_action :login_required
+  before_action :check_admin
+
+  def view_applications
```

Comment:
These controller actions should follow Rails RESTful protocol. Since you're returning all admin applicants, this should be the `index` action.